src/task.c

```diff
@@ -917,6 +918,50 @@ JL_DLLEXPORT jl_task_t *jl_new_task(jl_function_t *start, size_t ssize)
     return t;
 }
 
+JL_DLLEXPORT jl_task_t *jl_copy_task(jl_task_t *t)
+{
+    jl_task_t *newt = (jl_task_t*)jl_gc_allocobj(sizeof(jl_task_t));
+    jl_set_typeof(newt, jl_task_type);
+
+    newt->current_module = t->current_module;
+    newt->state = t->state;
+    newt->start = t->start;
+    newt->tls = jl_nothing; // does this point to newt.storate?
+    newt->result = t->result;
+    newt->parent = t->parent;
+    newt->consumers = jl_nothing;
+    newt->donenotify = jl_nothing;
+    newt->exception = jl_nothing;
+    newt->backtrace = jl_nothing;
+    newt->eh = NULL;
+    newt->gcstack = t->gcstack;
+    newt->tid = t->tid;
+
+    memcpy((void*)newt->ctx, (void*)t->ctx, sizeof(jl_jmp_buf));
+#ifdef COPY_STACKS
+    if (t->stkbuf){
+        newt->ssize = t->ssize;  // size of saved piece
+        newt->bufsz = t->bufsz;
+        newt->stkbuf = allocb(t->bufsz);
```

Comment:
The Julia [documentation](http://docs.julialang.org/en/release-0.4/manual/embedding/) page says `JL_GC_PUSH` is needed when performing `jl_...` calls.  Since there is `jl_...` call inside this function, is GC root still needed? Thanks!